clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -334,6 +334,13 @@ public OffsetResetStrategy resetStrategy(TopicPartition partition) {
         return assignedState(partition).resetStrategy;
     }
 
+    public boolean hasAllFetchPositions(Collection<TopicPartition> partitions) {
+        for (TopicPartition partition : partitions)
+            if (!hasValidPosition(partition))
+                return false;
+        return true;
+    }
+
     public boolean hasAllFetchPositions() {
```

Comment:
It's being used in a couple of place: [here](https://github.com/apache/kafka/blob/d1733953742236e0430aff4e502bd8ee7c642591/clients/src/test/java/org/apache/kafka/clients/consumer/internals/SubscriptionStateTest.java#L51) and [here](https://github.com/apache/kafka/blob/d1733953742236e0430aff4e502bd8ee7c642591/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L1043).

Even though they can be modified to leverage the new method, I'd prefer to keep it and just modify its implementation to delegate to the other one as you suggested.